### PR TITLE
test: extend core package unit tests [SPA-2030]

### PR DESCRIPTION
**Pull request by SofiaMargariti**
## Purpose

Extend unit test coverage in core package.

### CODE REVIEW

File: packages/core/src/utils/pathSchema.spec.ts [hunk 1]
```diff
@@ -0,0 +1,73 @@
+import {
+  isDeepPath,
+  lastPathNamedSegmentEq,
+  parseDataSourcePathIntoFieldset,
+  parseDataSourcePathWithL1DeepBindings,
```
Comment (Spring3) [inline packages/core/src/utils/pathSchema.spec.ts:5]: I wish we could use this function from our web app
https://github.com/contentful/user_interface/blob/master/src/javascripts/features/experience-builder/core/schema/utils.ts#L38-L55

It tackles both - deep and shallow bindings

Maybe as a future improvement

**Review by Spring3 — APPROVED** [review]
(Great job! Qululululluulu)
  v
🦃

**Review by elylucas — APPROVED** [review]
Good stuff, thanks Sofia!

File: packages/core/src/utils/pathSchema.spec.ts (hunk 1 above)
Comment (SofiaMargariti) [reply packages/core/src/utils/pathSchema.spec.ts:5]: Good suggestion! Maybe we could move this from the web app here and then export it and use it in both places.

**Pull request merged**